src/api/resources/rest/sample.ts

```diff
@@ -35,16 +31,8 @@ export class SampleResource extends RESTResource<types.Sample> implements Permis
         return this.connection.get<boolean>(this.getListMethodPath('slug_exists'), { name: slug });
     }
 
-    /**
-     * Use this method carefully. Check to make sure you need unannotated and annotated samples.
-     */
-    public query(query: types.QueryObject & AnnotationQuery, options?: QueryOptions): Rx.Observable<types.Sample[]> {
-        const {annotation, ...remainingQuery} = query;
-        const annotationQuery = annotation === 'annotated' ? { descriptor_completed: true } :
-                              annotation === 'unannotated' ? { descriptor_completed: false }
-                                                           : {};
-
-        return super.query({ ...remainingQuery, ...annotationQuery }, options);
+    public query(query: types.QueryObject, options?: QueryOptions): Rx.Observable<types.Sample[]> {
+        return super.query(query, options);
```

Comment:
Is this and `queryOne` override needed because of `query: types.QueryObject` type? Return type gets correctly inferred because of generic.

Tried deleting both `query` and `queryOne` and there are no compile errors.

Reply:
Yep, seems unnecessary. Merged QueryObject and Query and removed such lines